src/runner.py

```diff
@@ -158,32 +161,33 @@ def get_info(self):
             )
         return runners
 
-    def reconcile(self, num_runners):
+    def reconcile(self, virt_type, quantity):
         """Bring runners in line with target."""
         active_runners = []
         stale_runners = []
         for runner in self.get_info():
-            if runner.is_active:
-                active_runners.append(runner)
-            else:
-                stale_runners.append(runner)
+            if runner.virt_type == virt_type:
```

Comment:
Now that I think about it, the current implementation will probably not scale with additional units. If the reconciliation method finds 4 containers and 4 virtual machines, it will not create new ones, no matter how many units were deployed. That does not sound like the expected behavior. 

I will test it further.

Reply:
I'm glad you caught that from the README.  I guess we can do whatever is necessary.

Reply:
Testing the code on more than one unit confirmed that units were operating on runners that did not belong to them (runners that were not running on the unit). This is fixed now. The scaling behavior is aligned with the documentation:
* when a unit is added, the unit spawns N containers and M virtual machines;
* when a unit is removed, only containers and virtual machines started by that unit are removed.